python/taichi/lang/sparse_matrix.py

```diff
@@ -0,0 +1,103 @@
+class SparseMatrix:
+    is_taichi_class = True
+
+    def __init__(self, n=None, m=None, sm=None):
+        if m is not None:
+            self.n = n
+            self.m = m if m else n
+            self.matrix = self.__create_sparse_matrix()
+        else:
+            self.n = sm.num_rows()
+            self.m = sm.num_cols()
+            self.matrix = sm
+
+    def __create_sparse_matrix(self):
+        from taichi.lang.impl import get_runtime
```

Comment:
Is it possible to put this at the top level (if there is no circular import issue)?

Reply:
When I move it at the top level, some errors occur. It might be circular import error.